casper/src/main/scala/coop/rchain/casper/SignDeployment.scala

```diff
@@ -0,0 +1,30 @@
+package coop.rchain.casper
+
+import coop.rchain.casper.protocol.DeployData
+import coop.rchain.crypto.{PrivateKey, PublicKey}
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.crypto.signatures._
+import com.google.protobuf.ByteString
+
+object SignDeployment {
+
+  private def fill(
+      deployData: DeployData
+  )(user: PublicKey, sig: Array[Byte], sigAlgorithm: String): DeployData =
+    deployData
+      .withUser(ByteString.copyFrom(user.bytes))
+      .withSig(ByteString.copyFrom(sig))
+      .withSigAlgorithm(sigAlgorithm)
+
+  private def clear(deployData: DeployData): DeployData =
+    fill(deployData)(PublicKey(Array.empty[Byte]), Array.empty[Byte], "")
+
+  def apply(key: PrivateKey, deployData: DeployData, alg: SignaturesAlg = Ed25519): DeployData = {
+    val toSign    = clear(deployData).toByteString.toByteArray
+    val hash      = Blake2b256.hash(toSign)
+    val signature = alg.sign(hash, key.bytes)
```

Comment:
Since the long term goal is to avoid using Array[Byte] but to use Private/PublicKey, please add an overload sign : (Array[Byte], PrivateKey) => Array[Byte]